casual/casual-test/src/main/java/se/laz/casual/test/CasualEmbeddedServer.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * Copyright (c) 2024, The casual project. All rights reserved.
+ *
+ * This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+ */
+
+package se.laz.casual.test;
+
+import se.laz.casual.event.ServiceCallEvent;
+import se.laz.casual.event.ServiceCallEventStore;
+import se.laz.casual.event.ServiceCallEventStoreFactory;
+import se.laz.casual.event.server.EventServer;
+import se.laz.casual.event.server.EventServerConnectionInformation;
+import se.laz.casual.test.network.NetworkPortFactory;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * A light-weight instance of a casual server, initially focused on integration testing.
+ *
+ * NB - Currently provides just a casual event server interface.
+ */
+public class CasualEmbeddedServer
+{
+    private final UUID domainId;
+    private final Integer eventServerPort;
+    private final boolean eventServerEnabled;
+    private AtomicBoolean running = new AtomicBoolean( false );
+    private AtomicBoolean eventServerRunning = new AtomicBoolean( false );
+    private EventServer eventServer;
+
+    private CasualEmbeddedServer( Builder builder )
+    {
+        this.domainId = UUID.randomUUID();
+        this.eventServerPort = builder.eventServerPort;
+        this.eventServerEnabled = builder.eventServerEnabled;
+    }
+
+    /**
+     * Get the port for the event server instance.
+     *
+     * @return optional populated with port if provided otherwise empty.
+     */
+    public Optional<Integer> getEventServerPort()
+    {
+        return Optional.ofNullable( eventServerPort );
+    }
+
+    /**
+     * Is the event server enabled?
+     *
+     * @return if event server is enabled.
+     */
+    public boolean isEventServerEnabled()
+    {
+        return eventServerEnabled;
+    }
+
+    /**
+     * Is the event server currently running?
+     *
+     * @return if event server is running.
+     */
+    public boolean isEventServerRunning()
+    {
+        return eventServerRunning.get();
+    }
+
+    /**
+     * Is the casual embedded server running?
+     *
+     * @return if the embedded server is running.
+     */
+    public boolean isRunning()
+    {
+        return running.get();
+    }
+
+    /**
+     * Start the casual embedded server.
+     *
+     * NB - can only be called when the server is not running.
+     */
+    public void start()
+    {
+        if( running.compareAndSet( false, true ) )
+        {
+            initialiseEventServer();
+            return;
+        }
+        throw new IllegalStateException( "Server is already started." );
+    }
+
+    private void initialiseEventServer()
+    {
+        if( !this.eventServerEnabled )
```

Comment:
Yes it can, if you start the embedded server, but don't have the event server enabled, it wont start an event server. In future versions it will be more than just event server that is started when you start the embedded server.